tools/data_manipulation/xarray/xarray_select.xml

```diff
@@ -0,0 +1,288 @@
+<tool id="xarray_select" name="Netcdf Selection" version="0.15.1">
+    <description>extracts variable values with custom conditions on dimensions</description>
+    <requirements>
+        <requirement type="package" version="3">python</requirement>
+        <requirement type="package" version="1.5.3">netcdf4</requirement>
+        <requirement type="package" version="0.15.1">xarray</requirement> 
+        <requirement type="package" version="0.7.0">geopandas</requirement>
+        <requirement type="package" version="1.7.0">shapely</requirement> 
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+    mkdir output_dir &&
+    python '$__tool_directory__/xarray_tool.py' '$input' --select '$var' 
+           --verbose
+           --filter
+    #for $i,$uc in enumerate($user_choice)
+        #if $uc.condi_between.comparator=="bi"
+             '${uc.dim}#${uc.condi_between.comparator}#${uc.condi_between.t1}#${uc.condi_between.t2}'
+        #else
+             '${uc.dim}#${uc.condi_between.comparator}#${uc.condi_between.value}'
+        #end if
+    #end for
+
+    #if $time.condi_datetime.datetime=="yes"
+         --time 
+        #if $time.condi_datetime.condi_between.comparator=="sl"
+             '${time.condi_datetime.dim}#${time.condi_datetime.condi_between.comparator}#${time.condi_datetime.condi_between.t1}#${time.condi_datetime.condi_between.t2}'
+        #else
+             '${time.condi_datetime.dim}#${time.condi_datetime.condi_between.comparator}#${time.condi_datetime.condi_between.t1}'
+        #end if
+    #end if
+
+    #if $condi_source_coord.coord_source=="coord_from_file"
+         --coords '$coord_tabular'
+         --latname '$condi_source_coord.lat_dim' --lonname '$condi_source_coord.lon_dim'
+         --outputdir output_dir
+    #else
+         --outfile 'final.tabular'
+        #if $condi_source_coord.condi_coord.coord=='single'
+           --latname $condi_source_coord.condi_coord.lat_dim
+           --latvalN $condi_source_coord.condi_coord.lat_val
+           --lonname $condi_source_coord.condi_coord.lon_dim
+           --lonvalE $condi_source_coord.condi_coord.lon_val
+        #elif $condi_source_coord.condi_coord.coord=='subregion'
+           --latname $condi_source_coord.condi_coord.lat_dim
+           --latvalN $condi_source_coord.condi_coord.lat_valN
+           --latvalS $condi_source_coord.condi_coord.lat_valS
+           --lonname $condi_source_coord.condi_coord.lon_dim
+           --lonvalE $condi_source_coord.condi_coord.lon_valE
+           --lonvalW $condi_source_coord.condi_coord.lon_valW
+        #end if
+    #end if
+    ]]></command>
+    <inputs>
+        <param type="data" name="input" label="Input netcdf file" format="netcdf,h5"/>
```

Comment:
I guess I would enforce here netcdf ...

Reply:
Yes.That is probably best. I haven't changed what was in the original tool...